trie/iterator.go

```diff
@@ -232,7 +155,12 @@ func (it *NodeIterator) step() error {
 			}
 			for parent.child++; parent.child < len(node.Children); parent.child++ {
 				if current := node.Children[parent.child]; current != nil {
-					it.stack = append(it.stack, &nodeIteratorState{node: current, parent: ancestor, child: -1})
+					it.stack = append(it.stack, &nodeIteratorState{
+						hash:   common.BytesToHash(node.hash),
```

Comment:
I don't see why this should be set? The `hash` field supposed to be the hash of the current node, or `nil` if it's an internal node. Since we're iterating a full node, the children don't have hashes afaik. Also this code seems to set `hash` to the full node's hash, not the childs one. If I remove it though a test fails, which is even more odd, since the original code didn't have this and it passed.